velox/functions/lib/DateTimeFormatter.h

```diff
@@ -23,7 +23,18 @@
 
 namespace facebook::velox::functions {
 
-enum class DateTimeFormatterType { JODA, MYSQL, UNKNOWN };
+enum class DateTimeFormatterType {
+  JODA,
+  MYSQL,
+  // Corresponding to java.text.SimpleDateFormat in lenient mode. It is used by
+  // the 'date_format', 'from_unixtime', 'unix_timestamp' and
+  // 'to_unix_timestamp' Spark functions.
+  LENIENT_SIMPLE,
```

Comment:
nit: perhaps, add a TODO to clarify that this is currently no different from STRICT_SIMPLE